sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/processor/CompactBlockProcessor.kt

```diff
@@ -304,20 +305,29 @@ class CompactBlockProcessor internal constructor(
                         }
                         consecutiveChainErrors.set(0)
                         val napTime = calculatePollInterval()
-                        Twig.debug {
+                        Twig.info {
                             "$summary Sleeping for ${napTime}ms " +
                                 "(latest height: ${_processorInfo.value.networkBlockHeight})."
                         }
                         delay(napTime)
                     }
+                    is BlockProcessingResult.ContinuityError -> {
+                        Twig.error {
+                            "Failed while processing blocks at height: ${result.failedAtHeight} with continuity " +
+                                "error: ${result.error}"
+                        }
+                        // No nap time set to immediately continue with the following block synchronization attempt
+                    }
                     is BlockProcessingResult.SyncFailure -> {
                         Twig.error {
                             "Failed while processing blocks at height: ${result.failedAtHeight} with: " +
                                 "${result.error}"
                         }
-                        // TODO [#1222]: Enrich BlockProcessingResult.SyncFailure with root cause
-                        // TODO [#1222]: https://github.com/zcash/zcash-android-wallet-sdk/issues/1222
-                        checkErrorResult(result.failedAtHeight)
+                        val napTime = calculatePollInterval(true)
```

Comment:
Previously `checkErrorResult` would only retry a few times, and then would fail the sync process. AFAICT this no longer happens in this code pathway: for arbitrary non-continuity failures, we will now retry indefinitely (even if the errors are permanent). Is this desirable, or do we need to keep the old retry behaviour here?

The old behaviour also attempted to resolve the error by rewinding. This would only help with some kinds of recoverable errors, and we don't (and in some cases can't) distinguish between them, so I'm not against us just not trying to rewind, but indefinite retries without any rewinding seems like it won't be as useful as eventually telling the user "there's an error, you need to manually do something to fix it".

Reply:
That is true. There were more types of possible implementations of how to get out of the _rewind for every error_ problem. The previous one used a minimal changes approach with the _nap time_ implementation. But agree that it would change the SDK error-handling behavior. So I just pushed a change that adds fail logic after the `RETRIES` limit is reached, and thus keeps the same approach as before but does not rewind.